rustls/src/tls13/mod.rs

```diff
@@ -15,6 +15,9 @@ pub struct Tls13CipherSuite {
     /// How to compute HMAC with the suite's hash function.
     pub hmac_provider: &'static dyn crypto::hmac::Hmac,
 
+    /// How to complete HKDF with the suite's hash function.
+    pub hkdf_provider: &'static dyn crypto::tls13::Hkdf,
```

Comment:
It would be a slight confusion of the `tls13::Hkdf` interface to add it there, but on the other hand it would be a much simpler result than requiring the whole of `hmac::Hmac` when only a tiny proportion of it is actually used. Hmm.

Reply:
Have added a final commit which tries this. Seems a bit simpler, but I think it might need a bit of renaming of `trait Hkdf` to look sensible?

Reply:
Yeah, I like the changes but can't think of a better name either. Just leave it for now?